lib/rules/no-duplicate-landmark-labels.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const Rule = require('./base');
+const AstNodeInfo = require('../helpers/ast-node-info');
+
+function createErrorMessage(element) {
+  return `Duplicate label on <${element}> detected. If two of the same landmark elements or landmark roles are found in the same template, ensure that they have a unique label (provided by the aria-label or aria-labelledby attribute).`;
+}
+
+const ROLE_LANDMARK_MAP = {
+  banner: 'header',
+  main: 'main',
+  complementary: 'aside',
+  form: 'form',
+  search: 'form',
+  navigation: 'nav',
+  contentinfo: 'footer',
+};
+
+const LANDMARK_ELEMENTS = new Set(['header', 'main', 'aside', 'form', 'main', 'nav', 'footer']);
+
+const LANDMARK_ROLES = Object.keys(ROLE_LANDMARK_MAP);
+
+module.exports = class NoDuplicateLandmarkLabels extends Rule {
+  constructor(options) {
+    super(options);
+    this._landmarkHash = [];
+  }
+
+  visitor() {
+    return {
+      ElementNode(node) {
+        const hasRoleAttribute = AstNodeInfo.hasAttribute(node, 'role');
+
+        // Return if it is not a landmark element or doesn't have role
+        // or if the role value is not a string literal
+        if (
+          !(
+            hasRoleAttribute ||
+            LANDMARK_ELEMENTS.has(node.tag) ||
+            hasRoleAttribute.value.type === 'TextNode'
+          )
```

Comment:
```suggestion
          (
            hasRoleAttribute ||
            LANDMARK_ELEMENTS.has(node.tag) ||
            hasRoleAttribute.value.type === 'TextNode'
          ) === false
```

Multiline `if` conditionals can sometimes be hard for a developer to read. I think this one is fairly clear, but the `!` is hiding to reverse the entire check. Explicitly checking the evaluation to `=== false` may help avoid developer confusion of how this check works.